### ISSUE TITLE
IntelliSense not handling constexpr float types properly (starting from extension 1.17.2)

### ISSUE THREAD

**Issue by rsarrazin2** [microsoft/vscode-cpptools#11386]
### Environment

- OS and Version: Ubuntu 20.04 @ WSL2
- VS Code Version: 1.81.1
- C/C++ Extension Version: >= 1.17.2
- If using SSH remote, specify OS of remote machine: WSL2 @ Windows 11

### Bug Summary and Steps to Reproduce

Bug Summary: IntelliSense reports errors (at least on `constexpr` literals) that the compiler doesn't.

Steps to reproduce:
1. Clone the [minimum viable example](https://github.com/rsarrazin2/cmake_workspace_for_vscode_c_cpp_extension).
2. Build it using the CMake extension (works probably as well when manually running CMake).
3. Open [example.cpp](https://github.com/rsarrazin2/cmake_workspace_for_vscode_c_cpp_extension/blob/main/example/example/example.cpp).
4. If not happened automatically, run the "C/C++: Run Code Analysis on Active File" command.
5. You'll see the 4 IntelliSense errors that aren't reported by the compiler:

![image](https://github.com/microsoft/vscode-cpptools/assets/108058522/d7ebecaf-a87f-4d29-a671-7e37b884b100)

Expected behavior:
* No Intellisense errors shall be displayed.

Notes:
* Interestingly the errors affect both float and double literals, and the 0 literal doesn't show the error.
* Reverting to any version < 1.17.2 (e.g. 1.16.3) solves the problem.
* Issue opened upon suggestion by @Colengms in #11351, in [this comment](https://github.com/microsoft/vscode-cpptools/issues/11351#issuecomment-1698067031).

### Configuration and Logs

```shell
* No `c_cpp_properties.json` used
* No `C/C++: Log Diagnostics` command found
* Output from debug info with Debug log level: see file pasted in "Additional context"
```

### Additional context

[debug_output.log](https://github.com/microsoft/vscode-cpptools/files/12473247/debug_output.log)

**Comment by Colengms** [comment]
Note: This was moved from https://github.com/microsoft/vscode-cpptools/issues/11351#issuecomment-1697499243 , and the internal bug number is 1839628 (EDG).

**Comment by rsarrazin2** [comment]
Note that v1.18.5 still presents the issue.

**Comment by rsarrazin2** [comment]
This issue is about to celebrate its first birthday but is unfortunately still present in v1.21.6. Kindly requesting to reconsider for prioritization.

**Comment by sean-mcmanus** [comment]
@rsarrazin2 I've pinged the team.

**Comment by rsarrazin2** [comment]
The issue is still present in current release 1.22.11 and pre-release 1.23.2. I kindly remind that it used to work with revisions < 1.17.2 so that it should be possible to find the cause of the regression. Thank you.

**Comment by rsarrazin2** [comment]
Still present in 1.23.6. Searching issues for "expression must have a constant value" yields several hits (e.g. #12810, #12983) so it seems to be a real issue.

**Comment by ansjschuetz** [comment]
Any chance that this gets some attention?
Any workaround, apart from downgrading to 1.16.3..?

**Comment by sean-mcmanus** [comment]
@ansjschuetz Yeah, did you try 1.30.2 or the soon to be published 1.30.3? I was seeing it fixed, but it was with a different repro with the same errors, and I haven't confirmed it's fixed with your repro.

**Comment by sean-mcmanus** [comment]
Can anyone confirm if this still repros with https://github.com/microsoft/vscode-cpptools/releases/tag/v1.30.3 ?

**Comment by rsarrazin2** [comment]
I could bisect the fixing version to be [1.26.3](https://github.com/microsoft/vscode-cpptools/blob/main/Extension/CHANGELOG.md#version-1263-june-24-2025). I can confirm it also works as expected in the latest 1.30.3.

@ansjschuetz, if you confirm, I'd close the issue.

**Comment by ansjschuetz** [comment]
Thank you very much, @rsarrazin2 , for mentioning and asking me!
I just realised that an `extern ...` declaration was missing in my code.
With that, the problem is gone, so yes, please close the issue.
